### ISSUE TITLE
Activation not longer working

### ISSUE THREAD

**Issue by SergejFrank** [joshuaferrara/node-steam-key#3]
hey, 

i'm getting following error

node-steam-key/node_modules/bytebuffer/dist/bytebuffer-node.js:646
                throw RangeError("Illegal offset: 0 <= "+offset+" (+"+1+") <= "+this.buffer.length);

do u know how to fix that?

the activation of keys is not longer working

**Comment by joshuaferrara** [comment]
Please include the code that is causing this error, and I'll see if I can help.

**Comment by SergejFrank** [comment]
i just executed the example code

**Comment by joshuaferrara** [comment]
Will have to do some testing on this at a later date. Seems to be an issue with [decoding the purchase receipt info](https://github.com/joshuaferrara/node-steam-key/blob/master/index.js#L43).

**Comment by TheTimmaeh** [comment]
I've run into this, too. Would be nice if this gets fixed soon. I'll have a look myself too!

**Comment by joshuaferrara** [comment]
Here's the response packet. When attempting to decode as VDF, it seems to run into an issue. Wondering if the VDF format changed on the backend? I've included the ASCII version of the `purchase_receipt_info` and pasted the hex value below the screenshot.

![image](https://cloud.githubusercontent.com/assets/248680/24228531/1f7f9ed4-0f32-11e7-8f26-3cd9c595fed6.png)

```
purchase_receipt_info: 004D6573736167654F626A65637400075472616E73616374696F6E494400FFFFFFFFFFFFFFFF027061796D656E744D6574686F640001000000025075726368617365537461747573000200000002526573756C7444657461696C000E0000000A4261736550726963650000000000000000000A546F74616C446973636F756E740000000000000000000A5461780000000000000000000A7368697070696E670000000000000000000243757272656E6379436F64650000000000025472616E73616374696F6E54696D6500512CD358025061636B616765494400FFFFFFFF014572726F72486561646C696E650000014572726F72537472696E670000014572726F724C696E6B546578740000014572726F724C696E6B55524C00000808
```

**Comment by joshuaferrara** [comment]
VDF encoding still works, per verification of the rich presence API with my `node-csgo` library. Going to have to look into the decoding later, and see if there is something going wrong there.

**Comment by joshuaferrara** [comment]
Please see the vdf-workaround branch (52972b8) for what should be a temporary fix (I haven't been able to test it, rather just commented out the offending lines of code). Note though that any data in `purchase_receipt_info` will not be available for read so you'll just be blindly activating keys.

**Comment by jabbink** [comment]
Thanks, will look into that branch, as I also have this issue and it seems to be related to the VDF:

```
28 Mar 14:21:38 - connected to db and logged in on steam
28 Mar 14:21:38 - registering [some key here]
./node_modules/bytebuffer/dist/bytebuffer-node.js:646
                throw RangeError("Illegal offset: 0 <= "+offset+" (+"+1+") <= "+this.buffer.length);
                ^

RangeError: Illegal offset: 0 <= 414 (+1) <= 414
    at ByteBuffer.module.exports.ByteBufferPrototype.readUint8 (./node_modules/bytebuffer/dist/bytebuffer-node.js:646:23)
    at Object.exports.decode (./node_modules/node-steam-key/VDF.js:22:31)
    at Object.exports.decode (./node_modules/node-steam-key/VDF.js:31:44)
    at Object.exports.decode (./node_modules/node-steam-key/VDF.js:31:44)
    at Object.exports.decode (./node_modules/node-steam-key/VDF.js:31:44)
    at Object.exports.decode (./node_modules/node-steam-key/VDF.js:31:44)
    at Object.exports.decode (./node_modules/node-steam-key/VDF.js:31:44)
    at Object.exports.decode (./node_modules/node-steam-key/VDF.js:31:44)
    at Object.exports.decode (./node_modules/node-steam-key/VDF.js:31:44)
    at Object.exports.decode (./node_modules/node-steam-key/VDF.js:31:44)
```

**Comment by TheTimmaeh** [comment]
I guess Steam's response changed, since other tools (not written in node) have similar issues.

Sadly this got useless for me since I used to detect the items behind the keys.

**Comment by jabbink** [comment]
ah, I only use it to activate 50 CD keys on multiple accounts every 24h, and luckily I already know which item I'll receive.

**Comment by joshuaferrara** [comment]
Haven't had much time to work on this, nor do I foresee it in the future, so I've changed the default branch to `vdf-workaround` for now.

**Comment by SergejFrank** [comment]
Hey @joshuaferrara I tried to use BinaryKVParser instead of VDF and it looks fine for me.
maybe it is a better quick fix than removing purchase_receipt_info entirely

**Comment by joshuaferrara** [comment]
If anyone wants to implement [binarykvparser](https://www.npmjs.com/package/binarykvparser) as a required module, and make a PR, feel free. Currently working on finals, so no time for me.

**Comment by TheTimmaeh** [comment]
@SergejFrank Hey, could you drop the line you used to parse the result? That'd help a lot!

**Comment by joshuaferrara** [comment]
Fixed in 6c6633bd46730d9395595ac0a989613324b4f6af

Published 1.0.3 to npm at 8af3004
